### DISCUSSION TITLE
Is DB application consistent volume snapshots supported with cloudnative-pg?

### DISCUSSION THREAD

**Discussion by qdrddr** [cloudnative-pg/cloudnative-pg#3512] [Q&A] [answered]
Hi,
DB application consistency allows the push of all changes from memory to disks and stop of the write requests for a short time; during this window (typically a matter of seconds), a volume snapshot can capture application consistent backup (as opposed to crush-consistent backup) of the database, ensuring a _guaranteed_ restoration of backup from such a volume snapshot. Once the volume snapshot is completed, resume regular DB work. This short time typically is within a timeout period, so the application that utilizes this DB will not experience downtime. 

Is this available in cloudnative-pg?

**Comment by gbartolini** [comment, marked as answer]
It is all explained here: https://cloudnative-pg.io/documentation/current/backup_volumesnapshot/

In my opinion, CloudNativePG has one of the finest implementations of volume snapshot support for a Kubernetes database application. We do support both hot backup and cold backup, and just to be clear: none will create any downtime.

Indeed, given that cold backup is taken from a standby, you won't experience any downtime. However, even our hot backup implementation guarantees consistent backups as we ensure the WAL files between the start and stop of the backup are saved in the backup.

It might be worth looking at the recording of my recent talk at KubeCon 2023 about this very topic: https://www.youtube.com/watch?v=WGQq4MWzW6E&ab_channel=CNCF%5BCloudNativeComputingFoundation%5D

Finally, we are pioneering support for Kubernetes' VolumeGroupSnapshot feature (alpha), which will be released in the next few months.
